src/toil/fileStores/nonCachingFileStore.py

```diff
@@ -70,17 +70,18 @@ def open(self, job: Job) -> Generator[None, None, None]:
             # Finally delete the job from the worker
             os.remove(self.jobStateFile)
 
-    def writeGlobalFile(self, localFileName, cleanup=False):
+    def writeGlobalFile(self, localFileName: str, cleanup: bool=False) -> FileID:
         absLocalFileName = self._resolveAbsoluteLocalPath(localFileName)
         creatorID = self.jobDesc.jobStoreID
         fileStoreID = self.jobStore.writeFile(absLocalFileName, creatorID, cleanup)
         if absLocalFileName.startswith(self.localTempDir):
             # Only files in the appropriate directory should become local files
             # we can delete with deleteLocalFile
-            self.localFileMap[fileStoreID].append(absLocalFileName)
+            self.localFileMap[fileStoreID] = absLocalFileName
```

Comment:
This should not be changed. `localFileMap` can have multiple local files (in a list) for each file store ID. Changing this and `readGlobalFile` in this way I think are why the CI tests are failing with complaints that you are trying to delete `/`. The string you put in gets treated as a collection of (1-character) strings, one of which is `/`.

Reply:
Got it, just switched it back. I think I just got a bit mixed up from the typing for `localFileMap`.